### ISSUE TITLE
New Way of Loading Data

### ISSUE THREAD

**Issue by Programmer-RD-AI** [PyTorchLightning/pytorch-lightning#9844]
New Way of Loading Data

I think a new data loading way will be helpful

I think some help func like

[example.zip](https://github.com/PyTorchLightning/pytorch-lightning/files/7295517/example.zip)
s

With best regard

**Comment by Programmer-RD-AI** [comment]
`import os,cv2
import numpy as np
import torch,torchvision
from tqdm import tqdm

def load_data(data_dir:str,matrix:bool=False,idx_clf:bool=False):
    """
    data_dir : data_dir is the directory that you want to load the data from
    matrix : if you want y as [0,1,0]
    idx_clf : if you want y as 1 or 2 or 3
    """
    data = []
    labels = {}
    labels_r = {}
    idx = 0
    if matrix:
      for label in os.listdir(data_dir):
          idx += 1
          labels[label] = idx
          labels_r[idx] = label
      for folder in os.listdir(data_dir):
          for file in os.listdir(f'{data_dir}{folder}/'):
              img = cv2.imread(f'{data_dir}{folder}/{file}')
              img = cv2.resize(img,(56,56))
              img = img / 255.0
              data.append([
                  img,
                  np.eye(labels[folder]+1,len(labels))[labels[folder]-1]
              ])
      X = []
      y = []
      for d in data:
          X.append(d[0])
          y.append(d[1])
    if idx_clf:
      for folder in os.listdir(data_dir):
          idx += 1
          for file in os.listdir(f'{data_dir}{folder}/'):
              img = cv2.imread(f'{data_dir}{folder}/{file}')
              img = cv2.resize(img,(56,56))
              img = img / 255.0
              data.append([
                  img,
                  idx
              ])
      X = []
      y = []
      for d in data:
          X.append(d[0])
          y.append(d[1])
    return X,y,labels,labels_r,idx,data`

**Comment by Programmer-RD-AI** [comment]
import os,cv2
import numpy as np
import torch,torchvision
from tqdm import tqdm

def load_data(data_dir:str,matrix:bool=False,idx_clf:bool=False):
    """
    data_dir : data_dir is the directory that you want to load the data from
    matrix : if you want y as [0,1,0]
    idx_clf : if you want y as 1 or 2 or 3
    """
    data = []
    labels = {}
    labels_r = {}
    idx = 0
    if matrix:
      for label in os.listdir(data_dir):
          idx += 1
          labels[label] = idx
          labels_r[idx] = label
      for folder in os.listdir(data_dir):
          for file in os.listdir(f'{data_dir}{folder}/'):
              img = cv2.imread(f'{data_dir}{folder}/{file}')
              img = cv2.resize(img,(56,56))
              img = img / 255.0
              data.append([
                  img,
                  np.eye(labels[folder]+1,len(labels))[labels[folder]-1]
              ])
      X = []
      y = []
      for d in data:
          X.append(d[0])
          y.append(d[1])
    if idx_clf:
      for folder in os.listdir(data_dir):
          idx += 1
          for file in os.listdir(f'{data_dir}{folder}/'):
              img = cv2.imread(f'{data_dir}{folder}/{file}')
              img = cv2.resize(img,(56,56))
              img = img / 255.0
              data.append([
                  img,
                  idx
              ])
      X = []
      y = []
      for d in data:
          X.append(d[0])
          y.append(d[1])
    return X,y,labels,labels_r,idx,data

**Comment by Programmer-RD-AI** [comment]
so my idea is to have a matrix as the y ([0,1,0]) so this will be easy and I have seen some improvements rather than in indexing kind of y values. ([1],[0],[2])

I think this would be a great choice for people that use PyTorch lightning it will be easier for them.

I can help with creating this small helper function but I wanted to know the opinion from the people in the community about this, so is this even useful for them and other info.

With best regards.

If you have other questions please add a comment.
